### DISCUSSION TITLE
SPA? Svelte + pocketbase

### DISCUSSION THREAD

**Discussion by jramiroz98** [pocketbase/pocketbase#4353] [Q&A]
I really like how the admin Ui works
Wanted to know if there is a way to build inside of the same pocketbase file a public SPA using svelte, something similar to the admin route

I want a single file to rule them all
Thanks!

**Comment by ganigeorgiev** [comment]
You could build your Svelte SPA as usual but in order to embed it inside the binary you'll have to build it with Go (aka. [use PocketBase as framework](https://pocketbase.io/docs/go-overview/)) and use the [`//go:embed` directive](https://pkg.go.dev/embed).

For examples you can check the [Admin UI embed.go file](https://github.com/pocketbase/pocketbase/blob/master/ui/embed.go) or explore https://github.com/presentator/presentator where a Svelte SPA is embedded in similar manner.
